substrate/client/authority-discovery/src/worker.rs

```diff
@@ -347,8 +347,17 @@ where
 			self.client.as_ref(),
 		).await?.into_iter().collect::<HashSet<_>>();
 
-		if only_if_changed && keys == self.latest_published_keys {
-			return Ok(())
+		if only_if_changed {
+			// If the authority keys did not change and the `publish_if_changed_interval` was
+			// triggered then do nothing.
+			if keys == self.latest_published_keys {
+				return Ok(())
+			}
+
+			// We have detected a change in the authority keys, reset the timers to
+			// publish and gather data faster.
+			self.publish_interval.set_to_start();
```

Comment:
This would only detect that our keys have change, like when we rotate them and since we get here we are going to publish them, so what is the point of publishing them 2 seconds later ? 

Same for `query_interval` that is for the current node discovering other nodes, why we would try to discover more nodes just because our keys changed, the other nodes should actually try more often, not us.

Reply:
But I guess it can't hurt :)